File: src/main/java/fr/yodamad/svn2git/service/util/DateFormatter.java
package fr.yodamad.svn2git.service.util;

import static java.lang.String.format;

public abstract class DateFormatter {

    private static final String MS = "ms";
    private static final String SEC = "s";
    private static final String MIN = "min";

    public static String toNiceFormat(Long milliseconds) {
        if (milliseconds < 1000) return format("%s%s", milliseconds, MS);

        Long seconds = milliseconds / 1000;
        Long ms = milliseconds % 1000;

        if (seconds < 60) return format("%s%s %s%s", seconds, SEC, ms == 0 ? "" : ms, ms == 0 ? "" : MS);

        Long minutes = seconds / 60;
        seconds = seconds % 60;

        return format("%s%s%s%s%s%s", minutes, MIN,
            seconds == 0 ? "" : format(" %s", seconds), seconds == 0 ? "" : SEC,
            ms == 0 ? "" : format(" %s", ms), ms == 0 ? "" : MS);
    }
}

File: src/main/kotlin/fr/yodamad/svn2git/service/util/DateFormatter.kt
package fr.yodamad.svn2git.service.util

object DateFormatter {
    private const val MS = "ms"
    private const val SEC = "s"
    private const val MIN = "min"
    fun toNiceFormat(milliseconds: Long): String {
        if (milliseconds < 1000) return String.format("%s%s", milliseconds, MS)
        var seconds = milliseconds / 1000
        val ms = milliseconds % 1000
        if (seconds < 60) return String.format("%s%s %s%s", seconds, SEC, if (ms == 0L) "" else ms, if (ms == 0L) "" else MS)
        val minutes = seconds / 60
        seconds %= 60
        return String.format("%s%s%s%s%s%s", minutes, MIN,
            if (seconds == 0L) "" else String.format(" %s", seconds), if (seconds == 0L) "" else SEC,
            if (ms == 0L) "" else String.format(" %s", ms), if (ms == 0L) "" else MS)
    }
}
